framework/src/meshgenerators/PlaneIDMeshGenerator.C

```diff
@@ -63,7 +68,7 @@ PlaneIDMeshGenerator::generate()
 {
   std::unique_ptr<MeshBase> mesh = std::move(_input);
   if (mesh->mesh_dimension() < _axis_index + 1)
-    paramError("plane_axis", "PlaneIDMeshGenerator must operate on a proper plane axis");
+    paramError("plane_axis", "Plane axis must be contained in the mesh. Check mesh dimension");
```

Comment:
Do we want `mesh->spatial_dimension()` here?  `mesh->mesh_dimension()` can be 2 for e.g. a quad mesh that's in the XZ plane.

Reply:
good catch! Changed

Reply:
Reverted, the 3D test hits the error.

Reply:
Whoa, seriously?  We have a 3D mesh that's reporting spatial_dimension() less than 3?

Reply:
No it s reporting 2 for mesh dimension (even though it s real 3D) and that s what bypasses the error I think

Reply:
Nevermind i was wrong.

So it reports 3 for mesh dimension which is good because it s real 3D

And 1 for spatial dimension which is wrong. 

This is a dumb hex mesh that just got extruded from 2D to 3D

I think we need spatial_dimension to assert is_prepared (or do a reduction itself?). I just prepared the mesh and got the expected results of 3 and 3 for the dimensions

Reply:
We can't affordably recalculate there, but it would definitely be good to assert that the cache is still valid.  I'm reworking is_prepared() into finer-grained settings now, though, and I'll hold off until that's in shape so we can assert *just* that that cache is still valid.